patches/0004-Use-crypto-backends.patch

```diff
@@ -379,19 +380,129 @@ index 33942467784ad3..0282ffa9fa23c8 100644
++	AEAD
+ }
+ 
+ func (g gcmWithRandomNonce) NonceSize() int {
+@@ -157,7 +159,12 @@ func (g gcmWithRandomNonce) Seal(dst, nonce, plaintext, additionalData []byte) [
+ 		plaintext = ciphertext[:len(plaintext)]
+ 	}
+ 
+-	gcm.SealWithRandomNonce(g.GCM, nonce, ciphertext, plaintext, additionalData)
++	if _, ok := g.AEAD.(*aes.Block); ok {
++		gcm.SealWithRandomNonce(g.AEAD, nonce, ciphertext, plaintext, additionalData)
++	} else if boring.Enabled {
++		SealWithRandomNonce(g.AEAD, nonce, ciphertext, plaintext, additionalData)
++	}
++
+ 	return ret
+ }
+ 
+@@ -190,7 +197,7 @@ func (g gcmWithRandomNonce) Open(dst, nonce, ciphertext, additionalData []byte)
+ 		ciphertext = ciphertext[gcmStandardNonceSize:]
+ 	}
+ 
+-	_, err := g.GCM.Open(out[:0], nonce, ciphertext, additionalData)
++	_, err := g.AEAD.Open(out[:0], nonce, ciphertext, additionalData)
+ 	if err != nil {
+ 		return nil, err
+ 	}
+@@ -375,3 +382,32 @@ func sliceForAppend(in []byte, n int) (head, tail []byte) {
+ 	tail = head[len(in):]
+ 	return
+ }
++
++// SealWithRandomNonce encrypts plaintext to out, and writes a random nonce to
++// nonce. nonce must be 12 bytes, and out must be 16 bytes longer than plaintext.
++// out and plaintext may overlap exactly or not at all. additionalData and out
++// must not overlap.
++//
++// This complies with FIPS 140-3 IG C.H Scenario 2.
++//
++// Note that this is NOT a [cipher.AEAD].Seal method.
++func SealWithRandomNonce(g AEAD, nonce, out, plaintext, additionalData []byte) {
++	if uint64(len(plaintext)) > uint64((1<<32)-2)*gcmBlockSize {
++		panic("crypto/cipher: message too large for GCM")
++	}
++	if len(nonce) != gcmStandardNonceSize {
++		panic("crypto/cipher: incorrect nonce length given to GCMWithRandomNonce")
++	}
++	if len(out) != len(plaintext)+gcmTagSize {
++		panic("crypto/cipher: incorrect output length given to GCMWithRandomNonce")
++	}
++	if alias.InexactOverlap(out, plaintext) {
++		panic("crypto/cipher: invalid buffer overlap of output and input")
++	}
++	if alias.AnyOverlap(out, additionalData) {
++		panic("crypto/cipher: invalid buffer overlap of output and additional data")
++	}
++	// #TODO fips140.RecordApproved()
++	rand.Read(nonce)
++	g.Seal(out, nonce, plaintext, additionalData)
++}
 diff --git a/src/crypto/cipher/gcm_test.go b/src/crypto/cipher/gcm_test.go
-index e574822c9aa6f8..cf121d9e5f654d 100644
+index e574822c9aa6f8..8aa72e1877bba6 100644
 --- a/src/crypto/cipher/gcm_test.go
 +++ b/src/crypto/cipher/gcm_test.go
-@@ -8,7 +8,7 @@ import (
+@@ -8,7 +8,6 @@ import (
  	"bytes"
  	"crypto/aes"
  	"crypto/cipher"
 -	"crypto/internal/boring"
-+	boring "crypto/internal/backend"
  	"crypto/internal/cryptotest"
  	"crypto/internal/fips140"
  	fipsaes "crypto/internal/fips140/aes"
+@@ -727,9 +726,6 @@ func testGCMAEAD(t *testing.T, newCipher func(key []byte) cipher.Block) {
+ 
+ 			// Test NewGCMWithRandomNonce.
+ 			t.Run("GCMWithRandomNonce", func(t *testing.T) {
+-				if _, ok := block.(*wrapper); ok || boring.Enabled {
```

Comment:
Tests were not failing in your previous commit, which AFAIU it was not correct. I wouldn't merge this PR  without having a proper test that validates we are really generating a random nonce (which we should probably upstream).